src/screens/NoteListScreen.tsx

```diff
@@ -80,10 +80,17 @@ export default function NoteListScreen(props: PropsType) {
   const syncGate = useSyncGate();
   const theme = useTheme();
 
-  const colUid_ = props.route.params?.colUid;
-  const cacheCollection = (colUid_) ? cacheCollections.get(colUid_) : undefined;
-  // We only want to set colUid if a collection actually exists
-  const colUid = (cacheCollection) ? colUid_ : undefined;
+  const colUid = props.route.params?.colUid;
+  const cacheCollection = (colUid && colUid.length > 0) ? cacheCollections.get(colUid) : undefined;
+  
+  if (colUid && colUid.length > 0 && cacheCollection == null) {
+    navigation.navigate("404", {
```

Comment:
Maybe have a `navigateTo404(navigation, message="This notebook can't be found",  title="Not found")` utility function? So in most cases we can just call `navigateTo404(navigation)` and it'll work nicely? Having the text spread around everywhere feels very redundant.

Reply:
Sure, we can do that! I just thought at first the message would change more but it's mainly colUid in the params…